core/base/cinemaImaging/CinemaImaging.cpp

```diff
@@ -0,0 +1,35 @@
+#include <CinemaImaging.h>
+
+void ttk::CinemaImaging::multiplyByScalar(float *out,
+                                          const float *a,
+                                          const float &scalar) {
+  out[0] = a[0] + scalar;
+  out[1] = a[1] + scalar;
+  out[2] = a[2] + scalar;
+};
+
+void ttk::CinemaImaging::addVectors(float *out,
+                                    const float *a,
+                                    const float *b) {
+  out[0] = a[0] + b[0];
+  out[1] = a[1] + b[1];
+  out[2] = a[2] + b[2];
+};
+
+void ttk::CinemaImaging::subVectors(float *out,
+                                    const float *a,
+                                    const float *b) {
+  out[0] = b[0] - a[0];
+  out[1] = b[1] - a[1];
+  out[2] = b[2] - a[2];
+};
+
+void ttk::CinemaImaging::cross(float *out, const float *a, const float *b) {
```

Comment:
low level geometrical operations should be placed under `core/base/geometry/Geometry.h`.
note that these operations are likely to be available there already (this may imply merging/adapting code).

Reply:
It is OK if I'm the entry point for now, but Rosty wrote the entire BVH stuff and native renderer so I want to keep him as the author. I merged the cinema imaging code with all the other stuff (embree backend, vtk layer, etc) and "polished" everything. Separating these now again and opening separate PRs is overkill, so I would argue against it. I also agree that rudimentary functions should be moved to a more general class (or use them if they already exist). I would prefer if you merge this in as is and then we can open an issue requesting that, then I can tell Rosty to work on this.